### PR TITLE
Add courrelate zipkin consumer, split zipkin-prod into web/receiver

### PR DESCRIPTION
**Pull request by nickdella**
Add courrelate zipkin consumer, split zipkin-prod into web and receiver modules to speed up query performance. Code is a bit sloppy (cut/paste in build, courrelate consumer is not the cleanest), but this is in prototype mode as we build out courrelate. CC @danchia

### CODE REVIEW

File: project/Project.scala [hunk 1]
```diff
@@ -547,9 +549,57 @@ object Zipkin extends Build {
     receiverScribe, zookeeper
   )
 
-  lazy val prod = Project(
-    id = "zipkin-prod",
-    base = file("zipkin-prod"),
+  lazy val prodWeb = Project(
```
Comment (danchia) [inline project/Project.scala:552]: I think `zipkin-web` should basically what this does.

File: project/Project.scala [hunk 2]
```diff
@@ -547,9 +549,57 @@ object Zipkin extends Build {
...
+      libraryDependencies ++= Seq(
+        finagle("zipkin"),
+        finagle("stats"),
+        twitterServer
+      )
+    ).dependsOn(
+      web, query, cassandra, zookeeper
+    )
+
+  lazy val prodReceiver = Project(
```
Comment (danchia) [inline project/Project.scala:575]: `zipkin-collector-service` is very similar to this, although the main file needs modification to use kafka instead (https://github.com/twitter/zipkin/blob/29e0f90a8d19a6c80eaf0e7504f72058cd05b7a6/doc/kafka-receiver.md), so I'm ok with keeping this.

File: zipkin-courrelate/src/main/scala/org/coursera/zipkin/courrelate/CourrelateProducer.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import scala.collection.JavaConverters._
+
+class CourrelateProducer {
+  val logger = Logger.get(classOf[CourrelateProducer])
+
+  import Formats._
+
+  private[this] val producer = {
+    val downstreamKafkaBrokerList = System.getProperty("downstreamKafka.metadata.broker.list")
```
Comment (danchia) [inline zipkin-courrelate/src/main/scala/org/coursera/zipkin/courrelate/CourrelateProducer.scala:19]: maybe use the `flag` code that the rest of zipkin uses?

https://github.com/twitter/zipkin/blob/29e0f90a8d19a6c80eaf0e7504f72058cd05b7a6/zipkin-receiver-kafka/src/main/scala/com/twitter/zipkin/receiver/kafka/KafkaSpanReceiver.scala#L20

File: zipkin-courrelate/src/main/scala/org/coursera/zipkin/courrelate/CourrelateProducer.scala [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    val props = new Properties()
+    props.put("metadata.broker.list", downstreamKafkaBrokerList)
+    props.put("serializer.class", "kafka.serializer.StringEncoder")
+    props.put("request.required.acks", "1")
+    props.put("producer.type", "async")
+    props.put("compression.codec", "gzip")
```
Comment (danchia) [inline zipkin-courrelate/src/main/scala/org/coursera/zipkin/courrelate/CourrelateProducer.scala:26]: if cpu is a concern, maybe snappy is a better choice?

File: zipkin-courrelate/src/main/scala/org/coursera/zipkin/courrelate/Main.scala [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  root.setLevel(Level.INFO)
+  
+
+  import SpanUtils._
+
+  val producer = new CourrelateProducer()
+
+
+  def main(): Unit = {
+    var (totalReceived, totalServerReceived, totalProcessed) = (0, 0, 0)
```
Comment (danchia) [inline zipkin-courrelate/src/main/scala/org/coursera/zipkin/courrelate/Main.scala:46]: use atomics here? else you'll run into race conditions when you have multiple consumer threads

File: zipkin-courrelate/src/main/scala/org/coursera/zipkin/courrelate/SpanUtils.scala [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        val buffer = ann.value
+        val arr = new Array[Byte](buffer.remaining)
+        buffer.get(arr)
+        new String(arr, "UTF-8")
+      case AnnotationType.I32 => ann.value.asIntBuffer().get()
+      case AnnotationType.I64 => ann.value.asLongBuffer().get()
+      case _ => throw new IllegalArgumentException("Unhandled AnnotationType: " + ann.annotationType)
+    }
+
+    ann.value.rewind()
```
Comment (danchia) [inline zipkin-courrelate/src/main/scala/org/coursera/zipkin/courrelate/SpanUtils.scala:56]: will rewind work correctly if the byteBuffer started at a non-zero offset?

**Comment by danchia** [comment]
gave high-level feedback, style review elided since this is just prototype code :)

**Comment by nickdella** [comment]
Thanks for the feedback @danchia . Per our offline discussion, I'm closing this out in favor of an internal courrelate consumer.

**Pull request closed without merging**